src/main/java/frc/robot/subsystems/intake/IntakeIO.kt

```diff
@@ -73,4 +75,8 @@ abstract class IntakeIOReal(armMotorID: Int, rollersMotorID: Int, gearRatio:Doub
         rollerMotor.set(speed)
     }
 
+    override fun resetEncoder(currentPosition: Rotation2d) {
+        armEncoder.zeroOffset += currentPosition.radians
```

Comment:
The stowed position is 0.0, and motion towards the down position is positive. So this will actually just set the zero position to the current position, increasing the voltage applied to the arm at the hard stop position. You should be subtracting here, but definitely not by the current position, because that will set the "down" position to the stowed position, etc. You should subtract the (setPosition - CurrentPosition) from the zero offset.

Reply:
The point of this function is to set `zeroOffset` to a state where `currentPosition` is zero. I guess that might be bad naming on my part.

Reply:
Okay, but the way you are calling it still wont work. You are passing in the the angle to hard stop, which will behave as I described above.